mr/src/test/java/org/apache/iceberg/mr/TestCatalogs.java

```diff
@@ -256,6 +256,20 @@ public void testLoadCatalogHive() {
     Assertions.assertThat(hiveCatalog.get()).isInstanceOf(HiveCatalog.class);
   }
 
+  @Test
+  public void testLoadCustomCatalogWithHiveCatalogTypeSet() {
+    String catalogName = "barCatalog";
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogUtil.ICEBERG_CATALOG_TYPE),
+            CatalogUtil.ICEBERG_CATALOG_TYPE_HIVE);
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogProperties.CATALOG_IMPL),
+            CustomHadoopCatalog.class.getName());
+    conf.set(InputFormatConfig.catalogPropertyConfigKey(catalogName, CatalogProperties.WAREHOUSE_LOCATION),
+            "/tmp/mylocation");
+    Optional<Catalog> hiveCatalog = Catalogs.loadCatalog(conf, catalogName);
+    Assertions.assertThat(hiveCatalog.get()).isInstanceOf(CustomHadoopCatalog.class);
```

Comment:
should be `Assertions.assertThat(hiveCatalog).isPresent().get().isInstanceOf(CustomHadoopCatalog.class);`

Reply:
Fixed